lib/msf/core/post/dns/resolve_host.rb

```diff
@@ -0,0 +1,48 @@
+# -*- coding: binary -*-
+
+module Msf
+  class Post
+    module DNS
+      ###
+      #
+      # This module resolves session DNS
+      #
+      ###
+      module ResolveHost
+        # Takes the host name and makes use of nsloopup to resolve the IP
+        #
+        # @param [String] host Node for the ruby `send` method
+        # @return [String, nil] ip The resolved IP
+        def resolve_host(host)
+          ip = nil
+
+          if client.respond_to?(:net) && client.commands.include?(Rex::Post::Meterpreter::Extensions::Stdapi::COMMAND_ID_STDAPI_NET_RESOLVE_HOST)
+            result = client.net.resolve.resolve_host(host)
+
+            return if result[:ip].blank?
+
+            ip = result[:ip]
+          else
+            data = cmd_exec("nslookup #{host}")
+            if data =~ /Name/
+              # Remove unnecessary data and get the section with the addresses
+              returned_data = data.split(/Name:/)[1]
+              # check each element of the array to see if they are IP
+              returned_data.gsub(/\r\n\t |\r\n|Aliases:|Addresses:|Address:/, ' ').split(' ').each do |e|
+                if Rex::Socket.dotted_ip?(e)
```

Comment:
is this method meant to handle ipv6 addresses?

Reply:
Ah I had lifted that snippet from here:
https://github.com/rapid7/metasploit-framework/blob/a81a71c5dfb493bb478809d7335fc39d601c3881/modules/post/multi/gather/dns_srv_lookup.rb#L166-L181

I forgot to remove that.

Reply:
I pulled that snippet from here:
https://github.com/rapid7/metasploit-framework/blob/a81a71c5dfb493bb478809d7335fc39d601c3881/modules/post/multi/gather/dns_srv_lookup.rb#L166-L181

I looked at that method and assumed it was fine as it just checked `Checks to see if the supplied address is in “dotted” form` for ipv6 or ipv4.

Should I handle that differently then?